### ISSUE TITLE
`tsj` compilation is not consistent with `tsc` compilation

### ISSUE THREAD

**Issue by broofa** [vega/ts-json-schema-generator#1285]
**Expected behavior**:  If a project compiles cleanly with `tsc`, then `tsj.createProgram()` shouldn't fail.

**Observed behavior**:  In this [minimal, complete test case](https://replit.com/@broofa/tsj-livescript-error),  `npx tsc --noEmit` runs cleanly:
![CleanShot 2022-06-07 at 12 24 08@2x](https://user-images.githubusercontent.com/164050/172465302-3b21bb02-d107-46fc-859b-4c27d6bf36bc.png)

However, when attempting to compile a schema for `MyCustomType` using the code in `index.js`, it fails with a "Could not find a declaration file for module 'livescript'. "   (Note: This issue is fixed in `tsc` by adding the `declare module` declaration in types.d.ts, but apparently `tsj` doesn't pick up on that file being a thing?)

![CleanShot 2022-06-07 at 12 44 21@2x](https://user-images.githubusercontent.com/164050/172468793-b919775d-5142-4965-a951-6c9b55d1f7c1.png)

BTW, I've run into a similar issue with `tsj`'s module resolution logic in projects that use yarn workspaces.  If a dependency gets hoisted up out of a workspace directory and into the top-level repo directory, `tsj`'s module resolution won't find it.

Basically, `tsj`'s compilation logic should mirror whatever `tsc` is doing.  (And, yes, I know that's asking a lot.  I've used the `tsc` programmatic API and it's frustratingly low-level compared to what the `tsc` command line does.)

**Comment by broofa** [comment]
Note: I think I found the problem.  If specified, `tsj` uses the `config.path` file as the one `rootNames` entry that gets passed to `ts.createProgram()`, which is problematic if other files are required for compilation (such as supporting *.d.ts files).

The fix here is to just omit the `path` property when calling `tsj.createProgram` (`CONFIG.path` in my code above).  This causes `tsj` to fall back to the `fileNames` array that `ts` finds when parsing the tsconfig.json file.

**Comment by effulgentsia** [comment]
> The fix here is to just omit the path property

Thanks! I was stumped on this as well and this got me sorted out. In my case, however, I didn't want to run the schema generator on all files in my project, but the value of the `path` property can be a glob, so setting path to `{path/to/file1.ts,path/to/file2.d.ts}` worked. Leaving this comment here in case it helps someone else some day.
